### ISSUE TITLE
ContentPage with Constructor Injection leads to Exception

### ISSUE THREAD

**Issue by DrRxDev** [dotnet/maui#5376]
### Description

This bug #4072 has been fixed in preview 13 but now has reemerged in the current preview version 14. 

[Reproduction project](https://github.com/dotnet/maui/files/7852287/MauiDIShellBugSample.zip)

### Steps to Reproduce

Same as #4072.

### Version with bug

Preview 14 (current)

### Last version that worked well

Preview 13

### Affected platforms

Android

### Affected platform versions

Android 11

**Comment by jfversluis** [comment]
@DrRxDev did you see the note here: https://github.com/dotnet/maui/wiki/Migration-to-Preview-14#shell

> In order to use constructor injection with ContentPages in the context of Shell, you'll need to register them with DI just like you have your view models.

Does that help?

**Comment by DrRxDev** [comment]
No, I didn't see that. That fixes my issue. Thank you very much and sorry for my blindness!

**Comment by jfversluis** [comment]
No worries, it's coincidental that I knew about this one ;) Glad we got it sorted!
